mysticeti-core/src/commit_observer.rs

```diff
@@ -180,3 +180,47 @@ impl<H: ProcessedTransactionHandler<TransactionLocator> + Send + Sync> CommitObs
         self.commit_interpreter.committed = committed;
     }
 }
+
+#[derive(Default)]
+pub struct SimpleCommitObserver {
+    commit_interpreter: Linearizer,
+    committed_leaders: Vec<BlockReference>,
```

Comment:
Looking more at this PR - this should be a channel that we then will connect to sui, not a `Vec`?